backend/src/main/java/com/bakdata/conquery/models/query/statistics/NumberColumnStatsCollector.java

```diff
@@ -1,36 +1,62 @@
 package com.bakdata.conquery.models.query.statistics;
 
-import java.util.ArrayList;
-import java.util.Collection;
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.Comparator;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.atomic.AtomicLong;
-import java.util.function.BooleanSupplier;
 
-import com.bakdata.conquery.io.cps.CPSType;
+import c10n.C10N;
+import com.bakdata.conquery.models.i18n.I18n;
 import com.bakdata.conquery.models.query.PrintSettings;
 import com.bakdata.conquery.models.types.ResultType;
+import com.google.common.collect.Range;
 import lombok.Getter;
-import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import org.jetbrains.annotations.NotNull;
 
 @Getter
-public class NumberColumnStatsCollector<TYPE extends Number & Comparable<TYPE>> extends ColumnStatsCollector<Number> {
+@Slf4j
+public class NumberColumnStatsCollector<TYPE extends Number & Comparable<TYPE>> extends ColumnStatsCollector {
+
+	private final ResultType type;
 	private final DescriptiveStatistics statistics = new DescriptiveStatistics();
 	private final AtomicLong nulls = new AtomicLong(0);
 
-	private final List<TYPE> samples = new ArrayList<>();
 
+	private final Comparator<TYPE> comparator;
 
-	private final BooleanSupplier samplePicker;
+	private final NumberFormat formatter;
+	private final int expectedBins;
+	private final double upperPercentile;
+	private final double lowerPercentile;
 
-	private final Comparator<TYPE> comparator;
+	public NumberColumnStatsCollector(String name, String label, String description, ResultType type, PrintSettings printSettings, int expectedBins, double lowerPercentile, double upperPercentile) {
```

Comment:
Mich verwirrt, dass percentiles reingereicht werden und auch berechnet werden. Bitte einen anderen namen nehmen

Reply:
Was genau verwirrt dich daran? Und was muss anders benannt werden? Die percentilen werden verwendet um das Histogramm auf einen relevanten Wertebereich zu fokussieren.

Reply:
Mich hat verwirrt, dass die Percentilen durch den Constructor scheinbar von außen reingegeben werden, ob wohl sie nach meinem Verständnis erst aus dem `collect` dieser Klasse hervorgehen.

Ich schlage vor `lowerPercentile` -> `lowerBoundPercentileHint` zu nennen, damit klarer wird, dass auch die Daten einen Einfluss haben

Reply:
> Mich hat verwirrt, dass die Percentilen durch den Constructor scheinbar von außen reingegeben werden, ob wohl sie nach meinem Verständnis erst aus dem collect dieser Klasse hervorgehen.

Das hier reingegebene sind die percentilen die dann in describe verwendet werden, um dann die entsprechenden werte an der stelle auszuwerten. Sollten die Werte zu eng aneinander liegen werden sie expandiert.